src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java

```diff
@@ -33,37 +40,135 @@
+	"on block explode:",
+	"\tadd exploded blocks to {_exploded-blocks::*}",
+	"\t",
 	"\tloop exploded blocks:",
-	"\t\tadd loop-block to {exploded::blocks::*}"})
+	"\t\tif loop-block is sand:",
+	"\t\t\tset loop-block to stone",
+	"\t",
+	"\tset exploded blocks to blocks in radius 3 around event-location  # this will clear exploded blocks and set it to the specified blocks",
+	"\tremove all chests from exploded blocks",
+	"\tremove 3 stone from exploded blocks # will remove only 3 stone blocks",
+	"\tremove stone from exploded blocks # will remove all stones blocks - 'any' and 'all' will do the same",
+	"\tadd blocks in radius 3 around event-location to exploded blocks",
+	"\tdelete exploded blocks # will clear exploded blocks",
+	"\t",
+	"on entity explode:",
+	"\tremove all stone from exploded blocks",
+	"\t",
+	"on explode:",
+	"\tremove all stone from exploded blocks"})
 @Events("explode")
-@Since("2.5")
+@Since("2.5, INSERT VERSION (changing the list)")
 public class ExprExplodedBlocks extends SimpleExpression<Block> {
 
 	static {
-		Skript.registerExpression(ExprExplodedBlocks.class, Block.class, ExpressionType.COMBINED, "[the] exploded blocks");
+		Skript.registerExpression(ExprExplodedBlocks.class, Block.class, ExpressionType.SIMPLE, "[the] exploded blocks");
 	}
 	
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		if (!getParser().isCurrentEvent(EntityExplodeEvent.class)) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(BlockExplodeEvent.class, EntityExplodeEvent.class)) {
 			Skript.error("Exploded blocks can only be retrieved from an explode event.");
 			return false;
 		}
 		return true;
 	}
 	
-	@Nullable
 	@Override
+	@Nullable
 	protected Block[] get(Event e) {
-		List<Block> blockList = ((EntityExplodeEvent) e).blockList();
-		return blockList.toArray(new Block[blockList.size()]);
+		return getBlocks(e).toArray(new Block[0]);
+	}
+
+	@Override
+	@Nullable
+	public Iterator<? extends Block> iterator(Event e) {
+		List<Block> blocks = getBlocks(e);
+		return new ArrayList(blocks).iterator();
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(final ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case SET:
+			case DELETE:
+				return CollectionUtils.array(Block[].class);
+			case REMOVE:
+			case REMOVE_ALL:
+				return CollectionUtils.array(Block[].class, ItemType[].class);
+			case RESET:
+			default:
+				return null;
+		}
+	}
+	
+	@Override
+	public void change(Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+		List<Block> blocks = getBlocks(e);
+
+		switch (mode) {
+			case DELETE:
+			case SET:
+			case ADD:
+				if (mode != ChangeMode.ADD)
+					blocks.clear();
+					
+				if (mode != ChangeMode.DELETE) {
+					assert delta != null;
+					for (Object o : delta) {
+						if (o instanceof ItemType)
```

Comment:
This condition is always false due to impl of acceptChange, where ADD, SET and DELETE only allows Block[]